Remotion/Web/UnitTests/Core/WebStringTest.cs

```diff
@@ -0,0 +1,210 @@
+
+      Assert.That (webString.Type, Is.EqualTo (WebStringType.Text));
+      Assert.That (webString.GetRawValue(), Is.EqualTo (stringValue));
+    }
+
+    [Test]
+    public void CreateFromString_WithNullValue_ThrowsArgumentNullException ()
+    {
+      Assert.That (() => WebString.CreateFromString (null), Throws.ArgumentNullException);
+    }
+
+    [Test]
+    public void ImplicitConversionToString ()
+    {
+      var stringValue = "My <b>test</a> string";
+      WebString webString = stringValue;
+
+      Assert.That (webString.Type, Is.EqualTo (WebStringType.Text));
+      Assert.That (webString.GetRawValue(), Is.EqualTo (stringValue));
+    }
+
+    [Test]
+    public void IsEmpty ()
+    {
+      Assert.That (new WebString().IsEmpty, Is.True);
+      Assert.That (((WebString)"").IsEmpty, Is.True);
+      Assert.That (((WebString)" ").IsEmpty, Is.False);
+      Assert.That (((WebString)"test").IsEmpty, Is.False);
+    }
+
+    [Test]
+    public void Write_Text ()
+    {
+      var stringValue = "My <b>test</a>";
+      var webString = WebString.CreateFromString (stringValue);
+
+      var renderedString = ExecuteWithHtmlTextWriter (webString.Write);
+      Assert.That (renderedString, Is.EqualTo ("My &lt;b&gt;test&lt;/a&gt;"));
+    }
+
+    [Test]
+    public void Write_Html ()
+    {
+      var stringValue = "My <b>test</a>";
+      var webString = WebString.CreateFromRawHtml (stringValue);
+
+      var renderedString = ExecuteWithHtmlTextWriter (webString.Write);
+      Assert.That (renderedString, Is.EqualTo (stringValue));
+    }
+
+    [Test]
+    public void Write_Attribute_ThrowsInvalidOperationException ()
+    {
+      var stringValue = "My <b>test</a>";
+      var webString = WebString.CreateFromRawAttribute (stringValue);
+
+      Assert.That (() => ExecuteWithHtmlTextWriter (webString.Write), Throws.InvalidOperationException);
+    }
+
+    [Test]
+    public void AddAttribute_TextWithAttributeEnum ()
+    {
+      var stringValue = "My <b>test</a>";
+      var webString = WebString.CreateFromString (stringValue);
+
+      var renderedString = ExecuteWithHtmlTextWriter (
+          writer =>
+          {
+            webString.AddAttribute (writer, HtmlTextWriterAttribute.Class);
+            writer.RenderBeginTag ("a");
+            writer.WriteEndTag ("a");
+          });
+      Assert.That (renderedString, Is.EqualTo (@"<a class=""My &lt;b>test&lt;/a>""></a>"));
+    }
+
+    [Test]
+    public void AddAttribute_HtmlWithAttributeEnum_ThrowsInvalidOperationException ()
+    {
+      var stringValue = "My <b>test</a>";
+      var webString = WebString.CreateFromRawHtml (stringValue);
+
+      Func<string> testLambda = () => ExecuteWithHtmlTextWriter (writer => webString.AddAttribute (writer, HtmlTextWriterAttribute.Class));
+      Assert.That (testLambda, Throws.InvalidOperationException);
+    }
+
+    [Test]
+    public void AddAttribute_AttributeWithAttributeEnum ()
```

Comment:
Should test with an enum that encodes and one that doesnt.

Reply:
In another comment you said that we are missing the `fEncode: true`. What should the encode behavior with type `Text`?

Because we do not need another set of tests if we tell it to always encode.

Reply:
in the review-comment-view on github, it's a bit context-less.

Here, the relevant part is, that there's two kinds of enum values, those that auto-encode and those that don't. we need to show in the test that the correct behavior happens in both cases. It's okay to have a few extra tests if it just coveres the relevant permutations explicitly. Encoding can be tricky, we don't want to think about it.